### ISSUE TITLE
Why is there a big price difference between infracost and gcp calculator?

### ISSUE THREAD

**Issue by aliscott** [infracost/infracost#2473]
### Discussed in https://github.com/infracost/infracost/discussions/2472

<div type='discussions-op-text'>

<sup>Originally posted by **Alexlofern** May 23, 2023</sup>
<img width="342" alt="image" src="https://github.com/infracost/infracost/assets/27426979/0fc1fdd9-c2f0-47e9-8494-34e4272a4a0b">
<img width="232" alt="image" src="https://github.com/infracost/infracost/assets/27426979/403274e6-96d0-40c0-8c2a-b7e831fb3dc8">

I'm using 0.10.22 version.

Regards

Alexandre</div>

**Comment by aliscott** [comment]
@Alexlofern are you able to share the Terraform code that was used to generate the estimate?

**Comment by Alexlofern** [comment]
> @Alexlofern are you able to share the Terraform code that was used to generate the estimate?

Yes, sure
[main.tf.txt](https://github.com/infracost/infracost/files/11554911/main.tf.txt)
[output.tf.txt](https://github.com/infracost/infracost/files/11554912/output.tf.txt)
[provider.tf.txt](https://github.com/infracost/infracost/files/11554913/provider.tf.txt)
[variables.tf.txt](https://github.com/infracost/infracost/files/11554914/variables.tf.txt)

**Comment by Alexlofern** [comment]
> Thanks @Alexlofern looks like there's two issues here causing the mismatch.
> 
> 1. The Google pricing calculator estimate is for a zonal SQL instance, but the code is using a regional instance. You can change this in the Google pricing calculator by selecting 'Enable High Availability Configuration'.
> 2. Infracost is detecting the incorrect region, and currently showing the prices for us-central1. I've put in a fix for this here [fix(google): if region is specified at the resource level, use that #2475](https://github.com/infracost/infracost/pull/2475), which will be released in the next release. Until then, you should be able to get the correct estimate by specifying the region inline in the `provider` block, e.g:
> 
> ```
> provider "google" {
>   # Configuration options
>   region = "europe-west3"
> }
> ```

Hi, I did everything you asked, but there is still a small difference, I tried logged in and out of the calculator, the values are the same for my contract


<img width="377" alt="image" src="https://github.com/infracost/infracost/assets/27426979/8140aaf1-1a4e-4f0e-9d4e-fd63bb28ed9d">

<img width="241" alt="image" src="https://github.com/infracost/infracost/assets/27426979/a8100d03-6e95-4a9b-9f17-7064c624c211">

**Comment by aliscott** [comment]
Thanks @Alexlofern. I've been trying to reproduce running against the following code and have been unable to reproduce. I'm seeing the expected $22.48 cost:

```hcl
provider "google" {
  credentials = "{\"type\":\"service_account\"}"
  region      = "europe-west3"
}

resource "google_sql_database_instance" "sql_instance" {
  project             = "my-project"
  name                = "my-instance"
  region              = "europe-west3"
  database_version    = "POSTGRES_14"

  settings {
    tier = "db-f1-micro"
    availability_type = "REGIONAL"
    disk_size = 10

    ip_configuration {
      ipv4_enabled = false
    }
  }
}
```

It seems like somehow the resource is getting the wrong region, since those prices you are seeing are from `europe-west8` or `europe-west9` instead of `europe-west3`.

1. What command are you using the run `infracost`?
2. Does it show the correct price if you run with `INFRACOST_GOOGLE_OVERRIDE_REGION=europe-west3` env variable set?

**Comment by Alexlofern** [comment]
> Thanks @Alexlofern. I've been trying to reproduce running against the following code and have been unable to reproduce. I'm seeing the expected $22.48 cost:
> 
> ```terraform
> provider "google" {
>   credentials = "{\"type\":\"service_account\"}"
>   region      = "europe-west3"
> }
> 
> resource "google_sql_database_instance" "sql_instance" {
>   project             = "my-project"
>   name                = "my-instance"
>   region              = "europe-west3"
>   database_version    = "POSTGRES_14"
> 
>   settings {
>     tier = "db-f1-micro"
>     availability_type = "REGIONAL"
>     disk_size = 10
> 
>     ip_configuration {
>       ipv4_enabled = false
>     }
>   }
> }
> ```
> 
> It seems like somehow the resource is getting the wrong region, since those prices you are seeing are from `europe-west8` or `europe-west9` instead of `europe-west3`.
> 
> 1. What command are you using the run `infracost`?
> 2. Does it show the correct price if you run with `INFRACOST_GOOGLE_OVERRIDE_REGION=europe-west3` env variable set?

Hi, I just run:

        infracost breakdown --path $(SCRIPTRUNPATH) --format json --out-file $(SCRIPTRUNPATH)/infracost.json
        infracost output --path $(SCRIPTRUNPATH)/infracost.json --format table --show-skipped

I'll try use this var INFRACOST_GOOGLE_OVERRIDE_REGION, if have more suggestions, I appreciate.

Tks

**Comment by Alexlofern** [comment]
testing without replica, backup in the same region, same results:

<img width="399" alt="image" src="https://github.com/infracost/infracost/assets/27426979/fc6dce88-a582-4b50-8917-94a828e7c241">

<img width="231" alt="image" src="https://github.com/infracost/infracost/assets/27426979/864fe522-634b-428f-aa80-58deafa1d71f">

**Comment by aliscott** [comment]
> My main location is europe-west9, I use europe-west3 for replication of the database!

Thanks for the details. Just to clarify, is the following correct?
1. `var.LOCATION` is set to `europe-west9`
2. `var.LOCATION_DR` is set to `europe-west3`

If so then the expected prices should be:
* `google_sql_database_instance.sql_instance` to be $21.75 (price in `europe-west9`)
* `google_sql_database_instance.read_replica[0]` to be $22.48 (price in `europe-west3`)

Currently what you're seeing is:
* `google_sql_database_instance.sql_instance` is $21.75 (price in `europe-west9`)
* `google_sql_database_instance.read_replica[0]` is $21.75 (price in `europe-west9`)

**Comment by Alexlofern** [comment]
> > My main location is europe-west9, I use europe-west3 for replication of the database!
> 
> Thanks for the details. Just to clarify, is the following correct?
> 
> 1. `var.LOCATION` is set to `europe-west9`
> 2. `var.LOCATION_DR` is set to `europe-west3`
> 
> If so then the expected prices should be:
> 
> * `google_sql_database_instance.sql_instance` to be $21.75 (price in `europe-west9`)
> * `google_sql_database_instance.read_replica[0]` to be $22.48 (price in `europe-west3`)
> 
> Currently what you're seeing is:
> 
> * `google_sql_database_instance.sql_instance` is $21.75 (price in `europe-west9`)
> * `google_sql_database_instance.read_replica[0]` is $21.75 (price in `europe-west9`)

Yes, correct, tks

I found a glitch in my code and I fix

<img width="346" alt="image" src="https://github.com/infracost/infracost/assets/27426979/1039653a-d7b5-4cab-be91-07b014bff492">

<img width="233" alt="image" src="https://github.com/infracost/infracost/assets/27426979/3c979163-2d8e-4071-a805-3c3b9ba681b0">

**Comment by Alexlofern** [comment]
Oh, but when I deploy the second, I get the value from default of provider:

<img width="351" alt="image" src="https://github.com/infracost/infracost/assets/27426979/618d3571-5876-4725-a966-05ab1fb755d9">

**Comment by aliscott** [comment]
👍 thanks for confirming @Alexlofern, I'll update this issue when the release is out.

**Comment by aliscott** [comment]
@Alexlofern thanks for your help with this 🙏. This has now been released in [v0.10.23](https://github.com/infracost/infracost/releases/tag/v0.10.23)
